core/src/test/scala/io/delta/tables/DeltaTableBuilderSuite.scala

```diff
@@ -366,4 +368,85 @@ class DeltaTableBuilderSuite extends QueryTest with SharedSparkSession with Delt
         e.getMessage.contains("is not a valid name"))
     }
   }
+
+  test("delta table property case") {
+
+    val preservedCaseConfig = Map("delta.appendOnly" -> "true", "Foo" -> "Bar")
```

Comment:
Could you double check what will happen today if we have the same keys in different cases?

Reply:
Good catch, I extended the test to verify this use case